File: ruby/testcode/headerinj_008.rb
require_relative 'shared'

# vuln-code-snippet start ruby_headerinj_sanitize_crlf
def redirect_safe(req)
  url = req.param('url')
  safe_url = url.delete("\r\n") # vuln-code-snippet safe-line ruby_headerinj_sanitize_crlf
  BenchmarkResponse.new(302, '', { 'Location' => safe_url })
end
# vuln-code-snippet end ruby_headerinj_sanitize_crlf

File: php/testcode/headerinj_008.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_headerinj_multi_crlf
function headerinj008(BenchmarkRequest $req): BenchmarkResponse {
    $tracking = $req->header('X-Tracking-Id');
    header("X-Tracking-Id: " . $tracking); // vuln-code-snippet vuln-line php_headerinj_multi_crlf
    return BenchmarkResponse::ok('Tracked');
}
// vuln-code-snippet end php_headerinj_multi_crlf
